src/lib/operators/join_nested_loop.hpp

```diff
@@ -50,7 +76,10 @@ class JoinNestedLoop : public AbstractJoinOperator {
   std::shared_ptr<PosList> _pos_list_right;
 
   // for Full Outer, remember the matches on the right side
-  std::set<RowID> _right_matches;
+  std::vector<std::vector<bool>> _right_matches;
+
+  // The JoinIndex uses this join as a fallback if no index exists
+  friend class JoinIndex;
```

Comment:
Is it better using `friend` than making some of the functionality in the `JoinNestedLoop` public?

Reply:
Depends on how you define "better". Since no one else should use this, I think it's reasonable, but I don't care too much about it.